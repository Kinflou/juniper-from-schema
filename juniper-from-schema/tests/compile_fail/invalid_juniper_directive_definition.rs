#![allow(dead_code, unused_variables, unused_must_use, unused_imports)]
include!("../compile_pass/setup.rs");

juniper_from_schema::graphql_schema! {
    type Query {
        string: String!
    }

    schema { query: Query }

    directive @juniper(
        ownership: Boolean,
        infallible: String = "foo",
        with_time_zone: [String] = false,
        bar: [Boolean]
    ) on FIELD
}

pub struct Query;

impl QueryFields for Query {
    fn field_string<'a>(&self, executor: &Executor<'a, Context>) -> FieldResult<&String> {
        unimplemented!()
    }
}
