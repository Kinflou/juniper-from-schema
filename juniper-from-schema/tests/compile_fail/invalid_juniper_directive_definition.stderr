error: proc macro panicked
  --> $DIR/invalid_juniper_directive_definition.rs:4:1
   |
4  | / juniper_from_schema::graphql_schema! {
5  | |     type Query {
6  | |         string: String!
7  | |     }
...  |
16 | |     ) on FIELD
17 | | }
   | |_^
   |
   = help: message: 
           
           error: Invalid location for @juniper directive: `FIELD`
            --> schema:1:59
             |
           1 |    type Query { string : String ! } schema { query : Query } directive @ juniper
             |                                                              ^
           
           Location must be `FIELD_DEFINITION`
           
           
           error: Missing default value for `ownership` argument. Must be `"borrowed"`
            --> schema:2:2
             |
           2 |    (ownership : Boolean, infallible : String = "foo", with_time_zone : [String] =
             |     ^
           
           
           error: `ownership` argument must have type `String`
            --> schema:2:2
             |
           2 |    (ownership : Boolean, infallible : String = "foo", with_time_zone : [String] =
             |     ^
           
           Got `Boolean`
           
           
           error: Invalid default value for `infallible` argument. Must be `false`
            --> schema:2:23
             |
           2 |    (ownership : Boolean, infallible : String = "foo", with_time_zone : [String] =
             |                          ^
           
           Got `"foo"`
           
           
           error: `infallible` argument must have type `Boolean`
            --> schema:2:23
             |
           2 |    (ownership : Boolean, infallible : String = "foo", with_time_zone : [String] =
             |                          ^
           
           Got `String`
           
           
           error: Invalid default value for `with_time_zone` argument. Must be `true`
            --> schema:2:52
             |
           2 |    (ownership : Boolean, infallible : String = "foo", with_time_zone : [String] =
             |                                                       ^
           
           Got `false`
           
           
           error: `with_time_zone` argument must have type `Boolean`
            --> schema:2:52
             |
           2 |    (ownership : Boolean, infallible : String = "foo", with_time_zone : [String] =
             |                                                       ^
           
           Got `[String]`
           
           
           error: Invalid argument for @juniper directive: `bar`
            --> schema:3:9
             |
           3 |     false, bar : [Boolean]) on FIELD
             |            ^
           
           Supported arguments are `ownership`, `infallible`, and `with_time_zone`
           
           
           aborting due to 8 errors
